JASP-Desktop/qquick/datasetview.cpp

```diff
@@ -78,10 +78,15 @@ void DataSetView::setRolenames()
 
 }
 
-void DataSetView::modelDataChanged(const QModelIndex &, const QModelIndex &, const QVector<int> &)
+void DataSetView::modelDataChanged(const QModelIndex &index, const QModelIndex &, const QVector<int> &)
 {
-	modelAboutToBeReset();
-	calculateCellSizes();
+	QSizeF calcedSize = getTextSize(_model->data(index).toString());
+
+	if (_cellSizes[size_t(index.column())].width() < calcedSize.width())
```

Comment:
Now you are only comparing the current size of the cell with the new size of the cell. But the actual comparison should be made on the basis of the width of the whole column. 

Like: `_model->headerData(col, Qt::Orientation::Horizontal, _roleNameToRole["maxColString"]).toString();`
If you put that in a function you could reuse that easily, and then also use it at line 147

Reply:
OK?